src/confdb/converter/python/PythonConfigurationWriter.java

```diff
@@ -46,17 +56,24 @@ public String toString(IConfiguration conf, WriteProcess writeProcess) throws Co
 				+ converterEngine.getNewline());
 
 		str.append("import FWCore.ParameterSet.Config as cms\n\n");
-		if(conf.switchProducerCount() > 0) {
-		    str.append("from HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA\n\n");
-		}
-
+		
 
 		String object = "";
 		if (writeProcess == WriteProcess.YES) {
 			object = "process.";
 			str.append("process = cms.Process( \"" + conf.processName() + "\" )\n");
-		} else
+		} else {
 			indent = "";
+		}
+
+		if(conf.switchProducerCount() > 0) {
+			ReleaseVersionInfo relVarInfo = new ReleaseVersionInfo(conf.releaseTag());
+			str.append("\nfrom HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA\n\n");
+			if(relVarInfo.cycle()>=12 && relVarInfo.major()>=3){
```

Comment:
I'd say
```suggestion
			if(relVarInfo.cycle()>12 || relVarInfo.cycle()==12 && relVarInfo.major()>=3){
```
otherwise `CMSSW_13_0_0` will not get it.

Reply:
It's not an issue, bit I liked it better when the `import` were all grouped at the top (that's what `edmConfigDump` does), and only the `process.ProcessAcceleratorCUDA = ProcessAcceleratorCUDA()` part comes after the definition of the `process` itself.

Reply:
indeed, I also prefer that.  And was going to do it.  But then changed my mind because I would then need two separate loops which could get messy fast

https://github.com/cms-sw/hlt-confdb/blob/confdbv3-beta/src/confdb/converter/python/PythonConfigurationWriter.java#L69-L75

But reflecting on it, I could add all the lines needed to go after "process" to a temporary string which then I add later. So I'll do that as I agree its better to have the imports grouped

Reply:
> I'd say
> 
> otherwise `CMSSW_13_0_0` will not get it.

urgh, of course, thanks!